core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -42,13 +50,19 @@ class AddPartitionsToTxnManager(config: KafkaConfig, client: NetworkClient, time
   private val inflightNodes = mutable.HashSet[Node]()
   private val nodesToTransactions = mutable.Map[Node, TransactionDataAndCallbacks]()
 
+  private val metricsGroup = new KafkaMetricsGroup(this.getClass)
+  val verificationFailureRate = metricsGroup.newMeter("VerificationFailureRate", "failures", TimeUnit.SECONDS)
```

Comment:
You probably want to unregister these metrics when the thread is shutdown. Also, please add a test to validate that metrics are being removed correctly.

Reply:
That's a good call. Thanks